src/internal/ppsdb/ppsdb.go

```diff
@@ -25,21 +25,38 @@ var PipelinesVersionIndex = &col.Index{
 	Name: "version",
 	Extract: func(val proto.Message) string {
 		info := val.(*pps.PipelineInfo)
-		return VersionKey(info.Pipeline.Name, info.Version)
+		return VersionKey(info.Pipeline, info.Version)
 	},
 }
 
-func VersionKey(pipeline string, version uint64) string {
+// VersionKey return a unique key for the given project, pipeline & version.  If
+// the project is the empty string it will return an old-style key without a
+// project; otherwise the key will include the project.  The version is
+// zero-padded in order to facilitate sorting.
+func VersionKey(p *pps.Pipeline, version uint64) string {
+	projectName := p.Project.GetName()
+	pipelineName := p.Name
 	// zero pad in case we want to sort
-	return fmt.Sprintf("%s@%08d", pipeline, version)
+	if projectName == "" {
+		return fmt.Sprintf("%s@%08d", pipelineName, version) // pre-projects style
+	}
+	return fmt.Sprintf("%s/%s@%08d", projectName, pipelineName, version) // projects style
+}
+
+// PipelinesNameKey returns the key used by PipelinesNameIndex to index a
+// PipelineInfo.
+func PipelinesNameKey(p *pps.Pipeline) string {
+	if projectName := p.Project.GetName(); projectName != "" {
+		return fmt.Sprintf("%s/%s", projectName, p.Name)
```

Comment:
`p.String()`?

Reply:
Well, they are impliclty the same, but `Pipeline.String` is really meant to be human-readable first of all. Maybe it’s okay to use it?